src/publish/npm.coffee

```diff
@@ -40,10 +40,15 @@ module.exports = (dir, log) ->
 
         callback()
 
-  publish: (callback) ->
+  publish: (registry, callback) ->
     debug "publish"
 
-    exec "npm publish", options, (error, stdout, stderr) ->
+    registry = if registry
+      " --registry=#{registry}"
+    else
+      ''
+
+    exec "npm publish#{registry}", options, (error, stdout, stderr) ->
```

Comment:
Isn't this an `npm` feature? I'm not even sure if `npm` won't completely ignore `--registry` here because `publishConfig.registry` is set.